### ISSUE TITLE
📣 Help wanted..!

### ISSUE THREAD

**Issue by aichholzer** [GetRayo/rayo.js#40]
📣 Help would be greatly appreciated in these areas:

- [ ] 🔌 Writing plug-ins (as [briefly described](https://github.com/GetRayo/rayo.js/projects/1)) 
- [ ] 💬 Translating the docs (All languages are welcome) 
- [ ] 🏗 Improving (re-building) the site (https://rayo.js.org) 
- [ ] 🚚 Maintaining the site 
- [ ] 🔭 More (and better) unit and integration tests
- [ ] 🤳 Promotion 

🤔 Anything else you can think of..?
-Comments are welcome.
 
Anyone? 

Thank you!
👏👏👏

**Comment by waghanza** [comment]
Hi @aichholzer,

I have some ideas / contibutions for `rayo` :

+ Improving (re-building) the site (https://rayo.js.org)
   + Having the documentation over git (submodules or inside this repo) ?
   + Having a documentation build by a static generator ([nanoc](https://nanoc.ws), [Jekyll](https://jekyllrb.com))
   + Using an external service as documentation manager such as [read the docs](https://readthedocs.org)
+ Promotion
   + Using https://github.com/tbrand/which_is_the_fastest (when **production-ready**)

**Comment by QuadDepo** [comment]
I could look into building the website in React!

**Comment by aichholzer** [comment]
@QuadDepo -Thank you and that would be greatly appreciated. I was thinking about something like this https://poi.js.org/, simple, clean and to the point. Sections are clearly easy to navigate to from the menu bar and the information is what you need, not more, not less.

I know there are some nice documentation site templates available for React, but I just haven't had the time to dig into that. Creating something from scratch is also possible, but it might involve more work, of course.

[rayo.js](https://github.com/GetRayo/rayo.js) is a [monorepo](https://danluu.com/monorepo/) and I intend to keep it that way. Therefore, the best way to move forward would be for you to fork it and work in the `/docs` folder and when done just submit a PR to be merged.

Here some things the site should have:

- [ ] 100% mobile compatible
- [ ] stick to the [Rayo color pallete](https://github.com/GetRayo/Assets/blob/master/Color%20palette.pdf)
- [ ] easy navigation and clear section identifiers
- [ ] well coded and clear examples
- [ ] easily translate-able (I will translate it to english, german, spanish, dutch and italian)

Let me know if this is something you would want to look into and if there is anything else you need.
Happy building!
🏗

**Comment by A-Donato** [comment]
Hi! I can translate the docs to Spanish. Can you specify me the docs required?

**Comment by aichholzer** [comment]
@A-Donato 👏 👏  -For the time being, all I need translated is the `readme.md` and the `docs/index.html` files, since these are the ones that will get most exposure.
So, feel free to fork and submit a PR when you are done.

Thank you.
Much appreciated.

**Comment by A-Donato** [comment]
great! asap i will submit de pr (ง •̀_•́)ง

**Comment by motss** [comment]
Can I help in rebuilding the site with [Polymer](https://github.com/Polymer/polymer) instead?

**Comment by aichholzer** [comment]
@motss -Of course, 100%. I am always interested in the new stuff. I personally have no idea of Polymer, but I assume you do. What do you need from me to get it going?

**Comment by aichholzer** [comment]
Hey @gabrieleds keen on doing the `to italian` translation?

**Comment by gabrieleds** [comment]
@aichholzer I would suggest to skip translating the documentation:
- you will always require somebody to translate even the minor changes
- who works in IT knows English

**Comment by aichholzer** [comment]
Fair enough @gabrieleds -I get your point but if you were ever to travel to Spain or South America, you would come across many, many, many IT people how have a big lack of English and thus translated documents are always good.

**Comment by mcadamm4** [comment]
Hey @aichholzer, did anyone ever rebuild the site. I'm trying to learn react and would love to help out when I get a chance.

**Comment by aichholzer** [comment]
Hello @mcadamm4 -Thank you for reaching out. Sadly, in the end no one came through...
More than happy for you to work on it, either React or Vue would be great.

**Comment by QuadDepo** [comment]
@aichholzer @mcadamm4 I'm down! Do we have a slack channel for Rayo?

**Comment by aichholzer** [comment]
@QuadDepo -Slack; creating now

**Comment by mcadamm4** [comment]
@aichholzer @QuadDepo Thats amazing guys! Unfortunately (sort of), I'm going away on a trip for all of October, but I'll keep in touch and jump in when I get back.

**Comment by QuadDepo** [comment]
@aichholzer @mcadamm4 No worries! You gonna miss hacktober tho :(

I will try to setup a good base with input from @aichholzer, @mcadamm4 you got any react expierence already or just starting out?

**Comment by mcadamm4** [comment]
@QuadDepo Nope none, but I've been playing with vanilla JS for a couple months and have built a very small web app with Angular 4 before so I will be able to make a decent effort.

**Comment by aichholzer** [comment]
Alright gents, Slack is up. Please shot me an email (email address is on my profile) and I will send you an invitation in exchange. 🤝

**Comment by aichholzer** [comment]
@mcadamm4 & @QuadDepo -Added you both as collaborators; https://github.com/GetRayo/website

**Comment by aichholzer** [comment]
@vintagexav I was thinking about you... The React website might be something for you and where you can lend a hand. 👏

**Comment by vintagexav** [comment]
@aicholzer I would be glad to contribute

**Comment by Piroru** [comment]
Hey there!

@aichholzer A few question if you don't mind :)
- Is this repo still alive? 
- Are you planning to maintain it further? 
- Is slack channel created and active?

And here are my two cents:
- For me, it is not quite clear what makes Rayo faster than e.g. Express or Fastify. It would be great to have a few words in README.md about that. Of course it is routing at first, but still, I guess there are a few more things you can note
- React is a good stack for the site, but for making a good doc site from scratch with React it needs a time, and as you can see it is still in progress. So why are you not looking at solutions like [docusaurus](https://docusaurus.io/)(written with react) or [git book](https://www.gitbook.com/) (I'm a big fun of redux xD). Also maintaining react site is a bit complicated, I'm talking about time spending
- Typescript support would be great as well, at least @types/rayo package

Have a good coding  :)

**Comment by aichholzer** [comment]
Hello @Piroru -I absolutely do not mind the questions, here are some quick answers;

 > Is this repo still alive?

It certainly is. Alive and kicking. I am working on some new features at the moment, but lately my time is a bit limited.

 > Are you planning to maintain it further?

100%

> Is slack channel created and active?

There is a channel but by invitation only. It was created mainly to work around the new website.

Finally; I don't mean to sound arrogant, but how `Rayo` is faster than any other framework is actually irrelevant, the fact is that is the fastest and that should be the reason to use it or not. Everyone is welcome to read the code and understand how it works and what it does to achieve speed.

The intention of the new site was not to only show the documentation, but also related things. My plan for `Rayo` is to grow it into an eco-system that does more than just be an ultra-fast framework.

**Comment by waghanza** [comment]
@Piroru asking if `rayo` depends on what your faster idea is :stuck_out_tongue: 
You can see that `rayo` is no the fastest of the world, but very interesting and @aichholzer is working hard to that

You can see https://github.com/the-benchmarker/web-frameworks#latency for performance comparison

**Comment by Piroru** [comment]
@aichholzer @waghanza 
> You can see that rayo is no the fastest of the world, but very interesting and @aichholzer is working hard to that

Of course, it's not, and didn't say it, just talking about express/fastify comparison
> You can see https://github.com/the-benchmarker/web-frameworks#latency for performance comparison

Actually, this link helped me get here, so I've already been there :)
> Finally; I don't mean to sound arrogant, but how Rayo is faster than any other framework is actually irrelevant, the fact is that is the fastest and that should be the reason to use it or not. Everyone is welcome to read the code and understand how it works and what it does to achieve speed.

You absolutely right, but there is another point of view. It is good to clarify it for promotion, cos one day you gonna write for instance a medium article about how rayo is good, and you have to clarify it as well, otherwise, an article would be a bit odd, just my thoughts, it is up to you
> The intention of the new site was not to only show the documentation, but also related things. My plan for Rayo is to grow it into an eco-system that does more than just be an ultra-fast framework. At the end of the day, I am open to anything... Just looking for some help to grow this.

Your idea is awesome, but probably it is not really viable right now to have a full site right now. 
No hard feelings, but currently it is hard to read rayo.js.org, github README.md is much more convenient. But soon you'll have examples, plugins, reach API and so on, and you must migrate this base somewhere, cos README.md isn't really convinient. Anyway, my thoughts were to create a good documentation, get more collaborators and after that, you can decide do you need a react site or not, but again this is up to you.

**Comment by aichholzer** [comment]
@Piroru @waghanza -We could go on and argue about speed... Here are other, independent, benchmarks; https://github.com/fastify/benchmarks -`Rayo` is the fastest. ;) (`Rayo` can, of course, not be compared to frameworks written in more performant languages).

I agree with some sort of an article to deepen a bit on the concepts and ideas, I just need to find the time.

The full site is arguable, I guess anything is possible as long as we have the time and resources to do that. `rayo.js.org` is indeed a bit cumbersome to read and follow. Having a real good document site would be a great start. There is a `gitbook` page for `Rayo` but I still need to work that as well and get it nice and cleaned up.

In any case, I appreciate your comments and your feedback @Piroru 
👏
